src/mlpack/methods/reinforcement_learning/ppo.hpp

```diff
@@ -0,0 +1,154 @@
+/**
+ * @file ppo.hpp
+ * @author Xiaohong Ji
+ *
+ * This file is the definition of PPO class, which implements
+ * proximal policy optimization algorithms.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_RL_PPO_HPP
+#define MLPACK_METHODS_RL_PPO_HPP
+
+#include <mlpack/prereqs.hpp>
+
+#include "replay/random_replay.hpp"
+#include "mlpack/methods/ann/activation_functions/tanh_function.hpp"
+#include "mlpack/core/dists/gaussian_distribution.hpp"
+#include "training_config.hpp"
+
+namespace mlpack {
+namespace rl {
+
+/**
+ * Implementation of PPO algorithms. PPO is a new family of policy gradient
+ * methods for reinforcement learning, which alternate between sampling data
+ * through interaction with the environment, and optimizing a 'surrogate'
+ * objective function using stochastic gradient ascent.
+ *
+ * For more details, see the following:
+ * @code
+ * @article{Mnih2013,
+ *  author    = {John Schulman and
+ *               Filip Wolski and
+ *               Prafulla Dhariwal and
+ *               Alec Radford and
+ *               Oleg Klimov},
+ *  title     = {Proximal policy optimization algorithms},
+ *  journal   = {arXiv preprint arXiv:1707.06347},
+ *  year      = {2017},
+ *  url       = {https://arxiv.org/abs/1707.06347}
+ * }
+ * @endcode
+ *
+ * @tparam EnvironmentType The environment of the reinforcement learning task.
+ * @tparam NetworkType The network to compute action value.
+ * @tparam UpdaterType How to apply gradients when training.
+ * @tparam PolicyType Behavior policy of the agent.
+ * @tparam ReplayType Experience replay method.
+ */
+template <
+  typename EnvironmentType,
+  typename ActorNetworkType,
+  typename CriticNetworkType,
+  typename UpdaterType,
+  typename PolicyType,
+  typename ReplayType = RandomReplay<EnvironmentType>
+>
+class PPO
+{
+ public:
+  //! Convenient typedef for state.
+  using StateType = typename EnvironmentType::State;
+
+  //! Convenient typedef for action.
+  using ActionType = typename EnvironmentType::Action;
+
+  /**
+   * Create the PPO object with given settings.
+   *
+   * If you want to pass in a parameter and discard the original parameter
```

Comment:
I don't think it's needed to be specified here.  Though you can keep it, if you want to.

 Moreover, I would insist on giving a more explanatory description of PPO class above.